Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +390,178 @@ namespace Tensile
                 else
                     m_fanValues[i] += newValue;
             }
+            
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_freqValues[0]        = std::numeric_limits<uint16_t>::max();
+                m_powerValues[0]       = std::numeric_limits<uint16_t>::max();
+                m_temperatureValues[0] = std::numeric_limits<uint16_t>::max();
+            }
+            else
+            {
+                m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
```

Comment:
What is the return type of gpuMetrics.XXX?

Reply:
uint16_t, matches the array

Reply:
yes rsmi_dev_gpu_metrics_info_get returns false then before filling the m_freqValues I am checking if the index 0 has invalid value. I am using m_freqValues .resize(1) and push_back combination.

The nth_element function just rearranges the elements or does partial sorting to get median value.  

 gpuMetrics.XXX return value from the API is uint16_t